src/parsers/manifest/dash/common/indexes/template.ts

```diff
@@ -551,40 +564,37 @@ export default class TemplateRepresentationIndex implements IRepresentationIndex
     const { duration, timescale, endNumber, startNumber = 1 } = this._index;
 
     if (this._isDynamic) {
-      const lastPos = this._manifestBoundsCalculator.estimateMaximumBound();
-      if (lastPos === undefined) {
-        return undefined;
-      }
-      const agressiveModeOffset = this._aggressiveMode ? (duration / timescale) :
-                                                         0;
-      if (this._scaledRelativePeriodEnd !== undefined &&
+      const liveEdge = this._manifestBoundsCalculator.getEstimatedLiveEdge();
+      if (liveEdge !== undefined &&
+          this._scaledRelativePeriodEnd !== undefined &&
           this._scaledRelativePeriodEnd <
-            (lastPos + agressiveModeOffset - this._periodStart) * this._index.timescale) {
+            liveEdge - this._periodStart * this._index.timescale)
+      {
 
         let numberOfSegments = Math.ceil(this._scaledRelativePeriodEnd / duration);
         if (endNumber !== undefined && (endNumber - startNumber + 1) < numberOfSegments) {
           numberOfSegments = endNumber - startNumber + 1;
         }
         return (numberOfSegments - 1)  * duration;
       }
+      const maxPosition = this._manifestBoundsCalculator
+        .getEstimatedMaximumPosition(this._availabilityTimeOffset ?? 0);
+      if (maxPosition === undefined) {
+        return undefined;
+      }
+
       // /!\ The scaled last position augments continuously and might not
       // reflect exactly the real server-side value. As segments are
       // generated discretely.
-      const scaledLastPosition = (lastPos - this._periodStart) * timescale;
+      const scaledLastPosition = (maxPosition - this._periodStart) * timescale;
```

Comment:
is `maxPosition` a synonym for `lastPosition` ? 
I'm thinking it's a bit confusing to have two different words for the same thing

Reply:
I would guess so, yes.

OK I'm renaming it

Reply:
Done